site/components/pages/Pledge/PledgeForm/index.tsx

```diff
@@ -137,8 +142,12 @@ export const PledgeForm: FC<Props> = (props) => {
           id: "pledges.form.input.profileImageUrl.label",
           message: "Profile image url (optional)",
         })}
-        errors={formState.errors.profileImageUrl}
-        errorMessageMap={getPledgeFormErrorTranslations}
+        errorMessage={
+          !!formState.errors.profileImageUrl?.message &&
+          pledgeErrorTranslationsMap[
+            formState.errors.profileImageUrl.message as PledgeErrorId
+          ]
+        }
```

Comment:
aha very type safe.

i can live with the boilerplate but I would also be down for a helper fn to shave some lines, if you want

**edit: nvm this is harder than I thought because the error message mapping isn't uniform**

---

```ts
errorMessage={getFormError("profileImageUrl")}
```

it can just use the local state from the closure, I think
```ts
const getFormError = (form: string): string | undefined =>
    !!formState.errors[form]?.message ?
          pledgeErrorTranslationsMap[!!formState.errors[form].message as PledgeErrorId]
          : undefined;
```

Reply:
That's not so easy for dynamic field with FieldArrays (like the categories on the Pledge edit form).
I'll skip that for now